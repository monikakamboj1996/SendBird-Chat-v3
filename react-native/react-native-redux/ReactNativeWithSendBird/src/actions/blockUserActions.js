import { INIT_BLOCK_USER, BLOCK_LIST_SUCCESS, BLOCK_LIST_FAIL, USER_UNBLOCK_SUCCESS, USER_UNBLOCK_FAIL } from './types';
import { sbGetBlockUserList, sbUserUnblock } from '../sendbirdActions';

export const initBlockUser = () => {
  return { type: INIT_BLOCK_USER };
};

export const getBlockUserList = blockedUserListQuery => {
  return dispatch => {
    if (blockedUserListQuery.hasNext) {
      return sbGetBlockUserList(blockedUserListQuery)
        .then(blockedUsers => dispatch({ type: BLOCK_LIST_SUCCESS, list: blockedUsers }))
        .catch(error => dispatch({ type: BLOCK_LIST_FAIL, list: [] }));
    } else {
      dispatch({ type: BLOCK_LIST_FAIL, list: [] });
      return Promise.resolve(true);
    }
  };
};

export const onUnblockUserPress = unblockedUserId => {
  return dispatch => {
    return sbUserUnblock(unblockedUserId)
      .then(user => {
        dispatch({
          type: USER_UNBLOCK_SUCCESS,
          unblockedUserId: unblockedUserId
        });
      })
      .catch(error => dispatch({ type: USER_UNBLOCK_FAIL }));
  };
};
