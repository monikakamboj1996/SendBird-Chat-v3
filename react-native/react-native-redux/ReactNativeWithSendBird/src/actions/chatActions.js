import {
  INIT_CHAT_SCREEN,
  CREATE_CHAT_HANDLER_SUCCESS,
  CREATE_CHAT_HANDLER_FAIL,
  CHANNEL_TITLE_CHANGED,
  CHANNEL_TITLE_CHANGED_FAIL,
  MESSAGE_LIST_SUCCESS,
  MESSAGE_LIST_FAIL,
  SEND_MESSAGE_TEMPORARY,
  SEND_MESSAGE_SUCCESS,
  SEND_MESSAGE_FAIL,
  USER_BLOCK_SUCCESS,
  USER_BLOCK_FAIL,
  CHANNEL_EXIT_SUCCESS,
  CHANNEL_EXIT_FAIL,
  SEND_TYPING_START_SUCCESS,
  SEND_TYPING_START_FAIL,
  SEND_TYPING_END_SUCCESS,
  SEND_TYPING_END_FAIL,
  MESSAGE_RECEIVED,
  MESSAGE_UPDATED,
  MESSAGE_DELETED,
  CHANNEL_CHANGED,
  TYPING_STATUS_UPDATED,
  READ_RECEIPT_UPDATED,
  USER_MESSAGE_PRESS,
  USER_MESSAGE_SELECTION_CLEAR,
  OWN_MESSAGE_DELETED,
  OWN_MESSAGE_DELETED_FAIL,
  OWN_MESSAGE_UPDATED,
  OWN_MESSAGE_UPDATED_FAIL,
  MESSAGE_COPY
} from './types';

import {
  sbGetOpenChannel,
  sbOpenChannelEnter,
  sbGetChannelTitle,
  sbGetMessageList,
  sbSendTextMessage,
  sbSendFileMessage,
  sbUserBlock,
  sbGetGroupChannel,
  sbOpenChannelExit,
  sbTypingStart,
  sbTypingEnd,
  sbIsTyping,
  sbChannelDeleteMessage,
  sbChannelUpdateMessage,
  sbMarkAsRead
} from '../sendbirdActions';

import SendBird from 'sendbird';

export const initChatScreen = () => {
  const sb = SendBird.getInstance();
  sb.removeAllChannelHandlers();
  return { type: INIT_CHAT_SCREEN };
};

export const getChannelTitle = (channelUrl, isOpenChannel) => {
  return dispatch => {
    if (isOpenChannel) {
      return sbGetOpenChannel(channelUrl)
        .then(channel => {
          dispatch({
            type: CHANNEL_TITLE_CHANGED,
            title: sbGetChannelTitle(channel),
            memberCount: channel.participantCount
          });
        })
        .catch(error => {
          dispatch({ type: CHANNEL_TITLE_CHANGED_FAIL });
        });
    } else {
      return sbGetGroupChannel(channelUrl)
        .then(channel => {
          dispatch({
            type: CHANNEL_TITLE_CHANGED,
            title: sbGetChannelTitle(channel),
            memberCount: channel.memberCount
          });
        })
        .catch(error => {
          dispatch({ type: CHANNEL_TITLE_CHANGED_FAIL });
        });
    }
  };
};

export const createChatHandler = (channelUrl, isOpenChannel) => {
  return dispatch => {
    if (isOpenChannel) {
      return sbGetOpenChannel(channelUrl)
        .then(channel => {
          sbOpenChannelEnter(channel)
            .then(channel => {
              registerOpenChannelHandler(channelUrl, dispatch);
              dispatch({ type: CREATE_CHAT_HANDLER_SUCCESS });
            })
            .catch(error => dispatch({ type: CREATE_CHAT_HANDLER_FAIL }));
        })
        .catch(error => dispatch({ type: CREATE_CHAT_HANDLER_FAIL }));
    } else {
      return sbGetGroupChannel(channelUrl)
        .then(channel => {
          registerGroupChannelHandler(channelUrl, dispatch);
          dispatch({ type: CREATE_CHAT_HANDLER_SUCCESS });
        })
        .catch(error => dispatch({ type: CREATE_CHAT_HANDLER_FAIL }));
    }
  };
};

const registerCommonHandler = (channelHandler, channelUrl, dispatch) => {
  channelHandler.onMessageReceived = (channel, message) => {
    if (channel.url === channelUrl) {
      if (channel.isGroupChannel()) {
        sbMarkAsRead({ channel });
      }
      dispatch({
        type: MESSAGE_RECEIVED,
        payload: message
      });
    }
  };
  channelHandler.onMessageUpdated = (channel, message) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: MESSAGE_UPDATED,
        payload: message
      });
    }
  };
  channelHandler.onMessageDeleted = (channel, messageId) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: MESSAGE_DELETED,
        payload: messageId
      });
    }
  };
};

const registerOpenChannelHandler = (channelUrl, dispatch) => {
  const sb = SendBird.getInstance();
  let channelHandler = new sb.ChannelHandler();

  registerCommonHandler(channelHandler, channelUrl, dispatch);

  channelHandler.onUserEntered = (channel, user) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: CHANNEL_CHANGED,
        memberCount: channel.participantCount,
        title: channel.name
      });
    }
  };
  channelHandler.onUserExited = (channel, user) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: CHANNEL_CHANGED,
        memberCount: channel.participantCount,
        title: channel.name
      });
    }
  };

  sb.addChannelHandler(channelUrl, channelHandler);
};

const registerGroupChannelHandler = (channelUrl, dispatch) => {
  const sb = SendBird.getInstance();
  let channelHandler = new sb.ChannelHandler();
  registerCommonHandler(channelHandler, channelUrl, dispatch);
  channelHandler.onUserJoined = (channel, user) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: CHANNEL_TITLE_CHANGED,
        title: sbGetChannelTitle(channel),
        memberCount: channel.memberCount
      });
    }
  };
  channelHandler.onUserLeft = (channel, user) => {
    if (channel.url === channelUrl) {
      dispatch({
        type: CHANNEL_TITLE_CHANGED,
        title: sbGetChannelTitle(channel),
        memberCount: channel.memberCount
      });
    }
  };
  channelHandler.onReadReceiptUpdated = channel => {
    if (channel.url === channelUrl) {
      dispatch({ type: READ_RECEIPT_UPDATED });
    }
  };
  channelHandler.onTypingStatusUpdated = channel => {
    if (channel.url === channelUrl) {
      const typing = sbIsTyping(channel);
      dispatch({
        type: TYPING_STATUS_UPDATED,
        typing: typing
      });
    }
  };
  sb.addChannelHandler(channelUrl, channelHandler);
};

export const getPrevMessageList = previousMessageListQuery => {
  return dispatch => {
    if (previousMessageListQuery.hasMore) {
      return sbGetMessageList(previousMessageListQuery)
        .then(messages => {
          dispatch({
            type: MESSAGE_LIST_SUCCESS,
            list: messages
          });
        })
        .catch(error => dispatch({ type: MESSAGE_LIST_FAIL }));
    } else {
      dispatch({ type: MESSAGE_LIST_FAIL });
      return Promise.resolve(true);
    }
  };
};

export const onSendButtonPress = (channelUrl, isOpenChannel, textMessage) => {
  return dispatch => {
    if (isOpenChannel) {
      return sbGetOpenChannel(channelUrl)
        .then(channel => {
          sendTextMessage(dispatch, channel, textMessage);
        })
        .catch(error => dispatch({ type: SEND_MESSAGE_FAIL }));
    } else {
      return sbGetGroupChannel(channelUrl)
        .then(channel => {
          sendTextMessage(dispatch, channel, textMessage);
        })
        .catch(error => dispatch({ type: SEND_MESSAGE_FAIL }));
    }
  };
};

const sendTextMessage = (dispatch, channel, textMessage) => {
  const messageTemp = sbSendTextMessage(channel, textMessage, (message, error) => {
    if (error) {
      dispatch({ type: SEND_MESSAGE_FAIL });
    } else {
      dispatch({
        type: SEND_MESSAGE_SUCCESS,
        message: message
      });
    }
  });
  dispatch({
    type: SEND_MESSAGE_TEMPORARY,
    message: messageTemp
  });
};

export const onUserBlockPress = blockUserId => {
  return dispatch => {
    return sbUserBlock(blockUserId)
      .then(user => dispatch({ type: USER_BLOCK_SUCCESS }))
      .catch(error => dispatch({ type: USER_BLOCK_FAIL }));
  };
};

export const onUserMessagePress = message => {
  return { type: USER_MESSAGE_PRESS, message: message };
};

export const onMessageDelete = (channelUrl, isOpenChannel, message) => {
  return dispatch => {
    let promise = null;
    if (isOpenChannel) {
      promise = sbGetOpenChannel(channelUrl);
    } else {
      promise = sbGetGroupChannel(channelUrl);
    }
    return promise
      .then(channel => {
        sbChannelDeleteMessage(channel, message)
          .then(response => dispatch({ type: OWN_MESSAGE_DELETED }))
          .catch(error => dispatch({ type: OWN_MESSAGE_DELETED_FAIL }));
      })
      .catch(error => dispatch({ type: OWN_MESSAGE_DELETED_FAIL }));
  };
};

export const onUserUpdateMessage = (channelUrl, isOpenChannel, message, contents) => {
  return dispatch => {
    let promise = null;
    if (isOpenChannel) {
      promise = sbGetOpenChannel(channelUrl);
    } else {
      promise = sbGetGroupChannel(channelUrl);
    }
    return promise
      .then(channel => {
        return sbChannelUpdateMessage(channel, message, contents)
          .then(response => dispatch({ type: OWN_MESSAGE_UPDATED, edited: message, contents }))
          .catch(error => {
            dispatch({ type: OWN_MESSAGE_UPDATED_FAIL });
          });
      })
      .catch(error => {
        dispatch({ type: OWN_MESSAGE_UPDATED_FAIL });
      });
  };
};

export const onUserMessageCopy = () => {
  return { type: MESSAGE_COPY };
};

export const clearMessageSelection = () => {
  return { type: USER_MESSAGE_SELECTION_CLEAR };
};

export const onFileButtonPress = (channelUrl, isOpenChannel, source) => {
  return dispatch => {
    if (isOpenChannel) {
      return sbGetOpenChannel(channelUrl)
        .then(channel => {
          return sendFileMessage(dispatch, channel, source);
        })
        .then(message => {
          dispatch({
            type: SEND_MESSAGE_SUCCESS,
            message: message
          });
        })
        .catch(error => dispatch({ type: SEND_MESSAGE_FAIL }));
    } else {
      return sbGetGroupChannel(channelUrl)
        .then(channel => {
          return sendFileMessage(dispatch, channel, source);
        })
        .then(message => {
          dispatch({
            type: SEND_MESSAGE_SUCCESS,
            message: message
          });
        })
        .catch(error => {
          dispatch({ type: SEND_MESSAGE_FAIL });
        });
    }
  };
};

const sendFileMessage = (dispatch, channel, file) => {
  return new Promise((resolve, reject) => {
    sbSendFileMessage(channel, file, (message, error) => {
      if (error) {
        reject(error);
      } else {
        resolve(message);
      }
    });
  });
};

export const typingStart = channelUrl => {
  return dispatch => {
    return sbTypingStart(channelUrl)
      .then(response => dispatch({ type: SEND_TYPING_START_SUCCESS }))
      .catch(error => dispatch({ type: SEND_TYPING_START_FAIL }));
  };
};

export const typingEnd = channelUrl => {
  return dispatch => {
    return sbTypingEnd(channelUrl)
      .then(response => dispatch({ type: SEND_TYPING_END_SUCCESS }))
      .catch(error => dispatch({ type: SEND_TYPING_END_FAIL }));
  };
};

export const channelExit = (channelUrl, isOpenChannel) => {
  return dispatch => {
    if (isOpenChannel) {
      return sbGetOpenChannel(channelUrl)
        .then(channel => {
          sbOpenChannelExit(channel)
            .then(response => dispatch({ type: CHANNEL_EXIT_SUCCESS }))
            .catch(error => dispatch({ type: CHANNEL_EXIT_FAIL }));
        })
        .catch(error => dispatch({ type: CHANNEL_EXIT_FAIL }));
    } else {
      const sb = SendBird.getInstance();
      sb.removeChannelHandler(channelUrl);
      dispatch({ type: CHANNEL_EXIT_SUCCESS });
      return Promise.resolve(true);
    }
  };
};
