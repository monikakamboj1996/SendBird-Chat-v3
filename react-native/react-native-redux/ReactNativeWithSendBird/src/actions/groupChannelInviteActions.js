import {
  INIT_INVITE,
  USER_LIST_SUCCESS,
  USER_LIST_FAIL,
  CREATE_GROUP_CHANNEL_SUCCESS,
  CREATE_GROUP_CHANNEL_FAIL,
  INVITE_GROUP_CHANNEL_SUCCESS,
  INVITE_GROUP_CHANNEL_FAIL
} from './types';
import { sbGetUserList, sbGetGroupChannel, sbCreateGroupChannel, sbInviteGroupChannel } from '../sendbirdActions';

export const initInvite = () => {
  return { type: INIT_INVITE };
};

const getUserListFilter = (userListQuery, channelUrl, prevUsers, resolve, reject) => {
  if (!userListQuery.hasNext) {
    return resolve(prevUsers);
  }
  return sbGetUserList(userListQuery)
    .then(users => {
      if (channelUrl) {
        return sbGetGroupChannel(channelUrl)
          .then(channel => {
            const channelMemberIds = channel.members.map(member => {
              return member.userId;
            });
            const list = users.filter(user => {
              return channelMemberIds.indexOf(user.userId) < 0;
            });
            const filteredUsers = [...prevUsers, ...list];
            if (filteredUsers.length < userListQuery.limit / 2) {
              return getUserListFilter(userListQuery, channelUrl, filteredUsers, resolve, reject);
            } else {
              resolve(filteredUsers);
            }
          })
          .catch(() => {
            resolve([...users, ...prevUsers]);
          });
      } else {
        resolve([...users, ...prevUsers]);
      }
    })
    .catch(() => {
      resolve(prevUsers);
    });
};

export const getUserList = (userListQuery, channelUrl = null) => {
  return dispatch => {
    if (userListQuery.hasNext) {
      return new Promise((resolve, reject) => {
        const users = [];
        return getUserListFilter(userListQuery, channelUrl, users, resolve, reject);
      })
        .then(users => {
          dispatch({ type: USER_LIST_SUCCESS, list: users });
        })
        .catch(() => {
          dispatch({ type: USER_LIST_FAIL });
        });
    } else {
      dispatch({ type: USER_LIST_FAIL });
      return Promise.resolve(true);
    }
  };
};

export const createGroupChannel = (inviteUserIdList, isDistinct) => {
  return dispatch => {
    return sbCreateGroupChannel(inviteUserIdList, isDistinct)
      .then(channel => {
        dispatch({
          type: CREATE_GROUP_CHANNEL_SUCCESS,
          channel: channel
        });
      })
      .catch(error => {
        dispatch({ type: CREATE_GROUP_CHANNEL_FAIL });
      });
  };
};

export const inviteGroupChannel = (inviteUserIdList, channelUrl) => {
  return dispatch => {
    return sbInviteGroupChannel(inviteUserIdList, channelUrl)
      .then(channel => {
        dispatch({
          type: INVITE_GROUP_CHANNEL_SUCCESS,
          channel: channel
        });
      })
      .catch(error => {
        dispatch({ type: INVITE_GROUP_CHANNEL_FAIL });
      });
  };
};
