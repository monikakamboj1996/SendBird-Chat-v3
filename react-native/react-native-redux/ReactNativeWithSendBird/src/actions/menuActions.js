import { INIT_MENU, DISCONNECT_SUCCESS } from './types';
import { sbDisconnect } from '../sendbirdActions';

export const initMenu = () => {
  return { type: INIT_MENU };
};

export const sendbirdLogout = () => {
  return dispatch => {
    return sbDisconnect().then(() => dispatch({ type: DISCONNECT_SUCCESS }));
  };
};
