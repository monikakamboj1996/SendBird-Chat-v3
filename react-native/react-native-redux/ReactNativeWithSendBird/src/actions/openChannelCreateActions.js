import { INIT_OPEN_CHANNEL_CREATE, OPEN_CHANNEL_CREATE_SUCCESS, OPEN_CHANNEL_CREATE_FAIL } from './types';
import { sbCreateOpenChannel } from '../sendbirdActions';

export const initOpenChannelCreate = () => {
  return { type: INIT_OPEN_CHANNEL_CREATE };
};

export const createOpenChannel = channelName => {
  return dispatch => {
    return sbCreateOpenChannel(channelName)
      .then(channel =>
        dispatch({
          type: OPEN_CHANNEL_CREATE_SUCCESS,
          channel: channel
        })
      )
      .catch(error =>
        dispatch({
          type: OPEN_CHANNEL_CREATE_FAIL,
          error: error
        })
      );
  };
};
