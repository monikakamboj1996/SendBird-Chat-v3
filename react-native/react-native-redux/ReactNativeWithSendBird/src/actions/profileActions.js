import { INIT_PROFILE, GET_PROFILE_SUCCESS, UPDATE_PROFILE_SUCCESS, UPDATE_PROFILE_FAIL } from './types';
import { sbGetCurrentInfo, sbUpdateProfile } from '../sendbirdActions';

export const initProfile = () => {
  return { type: INIT_PROFILE };
};

export const getCurrentUserInfo = () => {
  return {
    type: GET_PROFILE_SUCCESS,
    userInfo: sbGetCurrentInfo()
  };
};

export const updateProfile = nickname => {
  return dispatch => {
    return sbUpdateProfile(nickname)
      .then(user => updateSuccess(dispatch, user))
      .catch(error => updateFail(dispatch, error));
  };
};

const updateFail = (dispatch, error) => {
  dispatch({
    type: UPDATE_PROFILE_FAIL,
    error: error
  });
};

const updateSuccess = dispatch => {
  dispatch({
    type: UPDATE_PROFILE_SUCCESS,
    userInfo: sbGetCurrentInfo()
  });
};
