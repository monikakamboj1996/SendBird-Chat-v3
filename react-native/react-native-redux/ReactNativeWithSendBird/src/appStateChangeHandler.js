export default (function() {
  var instance;

  function AppStateChangeHandler() {
    this.cbs = {};
    this.addCallback = (key, cb) => {
      this.cbs[key] = cb;
      return () => {
        delete this.cbs[key];
      };
    };

    this.notify = () => {
      for (let key in this.cbs) {
        this.cbs[key]();
      }
    };
  }

  return {
    getInstance: function() {
      if (!instance) {
        instance = new AppStateChangeHandler();
      }
      return instance;
    }
  };
})();
