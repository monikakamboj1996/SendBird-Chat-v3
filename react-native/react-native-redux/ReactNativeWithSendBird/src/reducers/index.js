import { combineReducers } from 'redux';
import login from './loginReducer';
import menu from './menuReducer';
import profile from './profileReducer';
import openChannel from './openChannelReducer';
import openChannelCreate from './openChannelCreateReducer';
import chat from './chatReducer';
import member from './memberReducer';
import blockUser from './blockUserReducer';
import groupChannel from './groupChannelReducer';
import groupChannelInvite from './groupChannelInviteReducer';

export default combineReducers({
  login,
  menu,
  profile,
  openChannel,
  openChannelCreate,
  chat,
  member,
  blockUser,
  groupChannel,
  groupChannelInvite
});

export loginReducer from './loginReducer';
export menuReducer from './menuReducer';
export profileReducer from './profileReducer';
export openChannelReducer from './openChannelReducer';
export openChannelCreateReducer from './openChannelCreateReducer';
export chatReducer from './chatReducer';
export memberReducer from './memberReducer';
export blockUserReducer from './blockUserReducer';
export groupChannelReducer from './groupChannelReducer';
export groupChannelInviteReducer from './groupChannelInviteReducer';
