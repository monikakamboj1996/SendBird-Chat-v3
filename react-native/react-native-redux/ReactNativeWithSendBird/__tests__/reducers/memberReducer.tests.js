import { memberReducer } from '../../src/reducers';
import { INIT_MEMBER, MEMBER_LIST_SUCCESS, MEMBER_LIST_FAIL } from '../../src/actions/types';

describe('reducers/memberReducer', () => {
  test('checking initial state', () => {
    const action = { type: 'nonexistent' };
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('INIT_MEMBER', () => {
    const action = { type: INIT_MEMBER };
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('MEMBER_LIST_SUCCESS', () => {
    const action = { type: MEMBER_LIST_SUCCESS, list: ['1', '2'] };
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });

  test('MEMBER_LIST_FAIL', () => {
    const action = { type: MEMBER_LIST_FAIL };
    expect(memberReducer(undefined, action)).toMatchSnapshot();
  });
});
