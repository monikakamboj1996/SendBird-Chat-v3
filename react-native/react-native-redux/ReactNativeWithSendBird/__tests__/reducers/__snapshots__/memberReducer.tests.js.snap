// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/memberReducer INIT_MEMBER 1`] = `
Object {
  "list": Array [],
}
`;

exports[`reducers/memberReducer MEMBER_LIST_FAIL 1`] = `
Object {
  "list": Array [],
}
`;

exports[`reducers/memberReducer MEMBER_LIST_SUCCESS 1`] = `
Object {
  "list": Array [
    "1",
    "2",
  ],
}
`;

exports[`reducers/memberReducer checking initial state 1`] = `
Object {
  "list": Array [],
}
`;
